src/lib/provider/Web3FunctionMultiChainProvider.ts

```diff
@@ -0,0 +1,28 @@
+import { ethers } from "ethers";
+import { Network } from "./types";
+import { ChainIdOfNetwork, Networks } from "./utils";
+
+export class Web3FunctionMultiChainProvider {
+  private _proxyRpcUrlBase: string | undefined;
+
+  constructor(proxyRpcUrlBase: string | undefined) {
+    this._proxyRpcUrlBase = proxyRpcUrlBase;
+  }
+
+  public network(network: Network): ethers.providers.StaticJsonRpcProvider {
+    const chainId = ChainIdOfNetwork[network];
+
+    if (this._isSupportedNetwork(network)) {
+      return new ethers.providers.StaticJsonRpcProvider(
+        `${this._proxyRpcUrlBase}/${chainId}`
+      );
```

Comment:
let's store the created provider in an internal map so that we can return it directly if accessed later again